File: app/src/main/java/com/cool/eye/mvp/java/test/TestPresenter.java
package com.cool.eye.mvp.java.test;


import com.cool.eye.mvp.java.IPresenterImpl;

/**
 * @author ycb
 * @date 2018/8/17
 */
public class TestPresenter extends IPresenterImpl<ITestView> {

    public TestPresenter(ITestView view) {
        super(view);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        loadDataFromDb();
    }

    @Override
    public void onResume() {
        super.onResume();
        loadDataFromNet();
    }

    private void loadDataFromDb() {
        //TODO
        getView().setTest("data");

    }

    private void loadDataFromNet() {
        //TODO
        getView().getLoadView().showLoading("loadDataFromNet");
        getView().setTest("data");
        getView().getLoadView().cancelLoading();
    }
}

File: app/src/main/java/com/cool/eye/mvp/kotlin/test/TestPresenter.kt
package com.cool.eye.mvp.kotlin.test

import com.cool.eye.mvp.kotlin.IPresenter

/**
 * @author ycb
 * @date 2018/8/17
 */
class TestPresenter(override val view: ITestView) : IPresenter<ITestView> {

    override fun onCreate() {
        super.onCreate()
        loadDataFromDb()
    }

    override fun onResume() {
        super.onResume()
        loadDataFromNet()
    }

    private fun loadDataFromDb() {
        //TODO
        view.setTest("data")

    }

    private fun loadDataFromNet() {
        //TODO
        view.showLoading("loadDataFromNet")
        view.setTest("data")
        view.cancelLoading()
    }
}